aiomysql/pool.py

```diff
@@ -132,6 +132,17 @@ def acquire(self):
 
     @asyncio.coroutine
     def _fill_free_pool(self, override_min):
+        # iterate over free connections and remove timeouted ones
+        free_size = len(self._free)
+        n = 0
+        while n < free_size:
```

Comment:
Hmm. Still not perfect: you are removing connection from deque and pushing it back if it is not closed (which is most likely happens).